core/chains/evm/chain.go

```diff
@@ -73,18 +140,50 @@ func (e errChainDisabled) Error() string {
 	return fmt.Sprintf("cannot create new chain with ID %s, the chain is disabled", e.ChainID.String())
 }
 
-func newTOMLChain(ctx context.Context, chain *toml.EVMConfig, opts ChainSetOpts) (*chain, error) {
+type GeneralConfig interface {
```

Comment:
/nit rename:

What is `GeneralConfig` general to? Should its name relate to why it's holding `config.AppConfig` and `toml.HasEVMConfigs`?

Reply:
This was just overlooked in the recent rename. It was `General`/`General`, and so should be:
```suggestion
type AppConfig interface {
```

I agree we should audit what needs whole app config regardless, but I suspect it is still necessary to support certain kinds of fallbacks.

Reply:
ah, that makes a bit more sense. updated the name and left the TODO